src/widgets/Survey/ui/PassingScreen.tsx

```diff
@@ -246,13 +255,21 @@ const PassingScreen = (props: Props) => {
     context.activityId,
     context.eventId,
     incrementStep,
+    hasNextSubStep,
+    handleNextSubStep,
     targetSubjectId,
     saveUserEventByType,
   ]);
 
   const onBack = useCallback(() => {
     saveUserEventByType('PREV', item);
 
+    // If the current item has subSteps and we're not at the first subStep
+    if (hasPrevSubStep) {
+      return handlePrevSubStep();
+    }
+
+    // If we're at the first subStep or there are no subSteps, go to the previous step
     if (!hasPrevStep) {
       return;
     }
```

Comment:
Or, we could invert the logic of this conditional and do:

```suggestion
    if (hasPrevStep) {
      return decrementStep({
	      activityId: context.activityId,
	      eventId: context.eventId,
	      targetSubjectId,
	    });
    }
    
    return;
```

Reply:
Makes sense